integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java

```diff
@@ -5,152 +5,185 @@
-        .timeout(getTimeout(nameValuePairList))
-        .apiKey(getNamedStringParameter(nameValuePairList, "api_key"))
-        .appName(getNamedStringParameter(nameValuePairList, "app_name"))
-        .urlParams(getUrlParams(nameValuePairList))
-        .consoleMode(false);
-
-    get(elements, "api", 1).ifPresent(builder::version);
-    get(elements, "namespaces", 3).ifPresent(builder::namespace);
-    get(elements, "jobs", 5).ifPresent(builder::jobName);
-    get(elements, "runs", 7).ifPresent(builder::parentRunId);
-  }
-
-  public static ArgumentParserBuilder builder() {
-    return new ArgumentParserBuilder() {
-      @Override
-      public ArgumentParser build() {
-        ArgumentParser argumentParser = super.build();
-        log.info(
-            String.format(
-                "%s/api/%s/namespaces/%s/jobs/%s/runs/%s",
-                argumentParser.getHost(),
-                argumentParser.getVersion(),
-                argumentParser.getNamespace(),
-                argumentParser.getJobName(),
-                argumentParser.getParentRunId()));
-        return argumentParser;
-      }
-    };
-  }
-
-  public static UUID getRandomUuid() {
-    return UUID.randomUUID();
+  @Builder.Default private String appName = null;
+  @Builder.Default private OpenLineageYaml openLineageYaml = new OpenLineageYaml();
+
+  public static ArgumentParser parse(SparkConf conf) {
+    ArgumentParserBuilder builder = ArgumentParser.builder();
+    adjustOldConfigs(conf);
+    conf.setIfMissing(SPARK_CONF_DISABLED_FACETS, DEFAULT_DISABLED_FACETS);
+    conf.setIfMissing(SPARK_CONF_TRANSPORT_TYPE, "http");
+
+    if (conf.get(SPARK_CONF_TRANSPORT_TYPE).equals("http")) {
+      findSparkConfigKey(conf, SPARK_CONF_HTTP_URL)
+          .ifPresent(url -> UrlParser.parseUrl(url).forEach(conf::set));
+    }
+    findSparkConfigKey(conf, SPARK_CONF_APP_NAME)
+        .filter(str -> !str.isEmpty())
+        .ifPresent(builder::appName);
+    findSparkConfigKey(conf, SPARK_CONF_NAMESPACE).ifPresent(builder::namespace);
+    findSparkConfigKey(conf, SPARK_CONF_JOB_NAME).ifPresent(builder::jobName);
+    findSparkConfigKey(conf, SPARK_CONF_PARENT_RUN_ID).ifPresent(builder::parentRunId);
+    builder.openLineageYaml(extractOpenlineageConfFromSparkConf(conf));
+    return builder.build();
   }
 
-  private static Optional<String> getNamedStringParameter(
-      List<NameValuePair> nameValuePairList, String name) {
-    return Optional.ofNullable(getNamedParameter(nameValuePairList, name))
-        .filter(StringUtils::isNoneBlank);
+  // adjust properties so the old pipelines are allowed
+  private static void adjustOldConfigs(SparkConf conf) {
+    findSparkConfigKey(conf, "spark.openlineage.host")
+        .ifPresent(
+            c -> {
+              replaceConfigEntry(conf, SPARK_CONF_HTTP_URL, c, "spark.openlineage.host");
+            });
+
+    findSparkConfigKey(conf, "spark.openlineage.timeout")
```

Comment:
are there any tests for this code?

Reply:
I've implemented some in https://github.com/OpenLineage/OpenLineage/blob/418cfb235547655351d72ab2929c96df23e71768/integration/spark/shared/src/test/java/io/openlineage/spark/agent/util/SparkConfUtilsTest.java#L19

Reply:
I think the proper tests should cover all the scenarios and for each deprecated key, it should be verified if it gets replaced with the new key.

Reply:
I checked and the entry with prefix test as well as this additional check for `.transport` is not needed as all the deprecated configurations are cleaned up in `adjustDeprecatedConfigs` in `ArgumentParser`